File: testdata/jennies/builders/foreign_builder/GoBuilder/builderpkg/somenicebuilder_builder_gen.go
package builderpkg

import (
	cog "github.com/grafana/cog/generated/cog"
	somepkg "github.com/grafana/cog/generated/somepkg"
)

var _ cog.Builder[somepkg.SomeStruct] = (*SomeNiceBuilderBuilder)(nil)

type SomeNiceBuilderBuilder struct {
    internal *somepkg.SomeStruct
    errors map[string]cog.BuildErrors
}

func NewSomeNiceBuilderBuilder() *SomeNiceBuilderBuilder {
	resource := &somepkg.SomeStruct{}
	builder := &SomeNiceBuilderBuilder{
		internal: resource,
		errors: make(map[string]cog.BuildErrors),
	}

	builder.applyDefaults()

	return builder
}

func (builder *SomeNiceBuilderBuilder) Build() (somepkg.SomeStruct, error) {
	var errs cog.BuildErrors

	for _, err := range builder.errors {
		errs = append(errs, cog.MakeBuildErrors("SomeNiceBuilder", err)...)
	}

	if len(errs) != 0 {
		return somepkg.SomeStruct{}, errs
	}

	return *builder.internal, nil
}

func (builder *SomeNiceBuilderBuilder) Title(title string) *SomeNiceBuilderBuilder {
    builder.internal.Title = title

    return builder
}

func (builder *SomeNiceBuilderBuilder) applyDefaults() {
}

File: testdata/jennies/builders/foreign_builder/TypescriptBuilder/src/builderPkg/someNiceBuilderBuilder.gen.ts
import * as cog from '../cog';
import * as somePkg from '../somePkg';

export class SomeNiceBuilderBuilder implements cog.Builder<somePkg.SomeStruct> {
    protected readonly internal: somePkg.SomeStruct;

    constructor() {
        this.internal = somePkg.defaultSomeStruct();
    }

    /**
     * Builds the object.
     */
    build(): somePkg.SomeStruct {
        return this.internal;
    }

    title(title: string): this {
        this.internal.title = title;
        return this;
    }
}

